### ISSUE TITLE
Handle unfetch() of records

### ISSUE THREAD

**Issue by cowboyd** [flexyford/impagination#6]
Add the ability to pass in an `unfetch` option that will be invoked whenever a page is unloaded. This allows you to do thing like perform any cleanup associated with those records such as unload the record from a data store:

```javascript
let dataset = new Dataset({
  fetch: function(pageOffset, pageSize) {
    return this.store.find('color', {page: pageOffset, perPage: pageSize});
  },
  unfetch: function(records) {
    records.forEach(function(record) {
      record.unload();
    });
  }
});

**Comment by flexyford** [comment]
Worked on this for #12 but implemented 
``` javascript
unfetch: function(pageOffset) {
    ...
  }
```
I'll edit the code to reflect your implementation above

**Comment by flexyford** [comment]
Fixed on #12
